tests/tasks/create_test_interfaces_with_dhcp.yml

```diff
@@ -43,8 +43,15 @@
       nmcli d set testbr managed false
     fi
     ip link set testbr up
-    ip addr add 192.0.2.1/24 dev testbr
-    ip -6 addr add 2001:DB8::1/32 dev testbr
+    timer=0
+    while ! ip a s testbr | grep -q 'inet [1-9]'
+    do
+        let "timer+=1"
+        if [ $timer -eq 30 ]; then break; fi
+        ip addr add 192.0.2.1/24 dev testbr
+        ip -6 addr add 2001:DB8::1/32 dev testbr
+        sleep 1
```

Comment:
why does it need a loop and a sleep? Can you not just add it, if it doesn't exist and proceed without sleep?

Reply:
Here is the racing problem, notice the IP address is not added without sleep:
```
[root@ci-vm-10-0-139-185 ~]# ip link set testbr up  && ip addr add 192.0.2.1/24 dev testbr && ip -6 addr add 2001:DB8::1/32 dev testbr
[root@ci-vm-10-0-139-185 ~]# ip a s testbr
98: testbr: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 42:d8:c3:44:1e:76 brd ff:ff:ff:ff:ff:ff
    inet6 fe80::40d8:c3ff:fe44:1e76/64 scope link
       valid_lft forever preferred_lft forever
[root@ci-vm-10-0-139-185 ~]# ip link set testbr up && sleep 1 && ip addr add 192.0.2.1/24 dev testbr && ip -6 addr add 2001:DB8::1/32 dev testbr
[root@ci-vm-10-0-139-185 ~]# ip a s testbr
98: testbr: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc noqueue state UP group default qlen 1000
    link/ether 42:d8:c3:44:1e:76 brd ff:ff:ff:ff:ff:ff
    inet 192.0.2.1/24 scope global testbr
       valid_lft forever preferred_lft forever
    inet6 2001:db8::1/32 scope global
       valid_lft forever preferred_lft forever
    inet6 fe80::40d8:c3ff:fe44:1e76/64 scope link
       valid_lft forever preferred_lft forever
```

Reply:
On slow machines (i.e. some of our downstream QE machines), there is a test failure because `ip link set testbr up` takes a few seconds

Reply:
> ok, whatever works !!

yeah - this is a case of 'just make it work in the test environment' . . .

Reply:
If `ip link set testbr up` takes a longer time, then it seems to me that the while loop should check that this finished instead of checking that the address is there.

Reply:
I think that the hard part will be checking the `ip link set testbr up` actually finishes before adding the addresses. I was trying to  use `while ! ip addr show testbr | grep -q 'state UP'` to validate the testbr status, but I still run into the same failure (IP addresses is not configured for the client side). But I will try again next week.

Reply:
For `ip link set testbr up`, ip link should be requesting an ACK from netlink to wait until the operation is completed so it should already be working without the sleep, it seems to me there is a bug somewhere

Reply:
> If `ip link set testbr up` takes a longer time, then it seems to me that the while loop should check that this finished instead of checking that the address is there.

I updated.